internal/parse/parser.go

```diff
@@ -6,6 +6,7 @@ import (
 	"strings"
 )
 
+// Parser keeps track of the current parsing state.
```

Comment:
This is not so helpful, in the sense that every struct type is there to keep track of its internal state, by definition.  As a rule I use for myself and recommend in general, we should only write comments when they are not obvious from context. This has two advantages: first, we don't waste people's time reading text that is not going to add any useful information, and second, we become more critical of our own comments so that they are indeed relevant and helpful. For example, there are probably good things to be said about what is the role of the Parser in this module.

Reply:
Noted, comment removed.